### PR TITLE
enable lock files for dui3

### PR DESCRIPTION
**Pull request by adamhathcock**
enable nuget lock files for dui3

- this will allow us to better monitor change in dependencies
- use lock files for restoration for repeatable builds
- cache nuget and restore for faster builds

### CODE REVIEW

**Review by AlanRynne — CHANGES_REQUESTED** [review]
I think we should also modify the CI to run in locked mode since we're at it

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,32 +12,24 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
-    outputs:
-      version: ${{ steps.set-version.outputs.version }}
+    runs-on: windows-latest
```
Comment (AlanRynne) [inline .github/workflows/ci.yml:15]: 💔

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -9,10 +9,47 @@ on:
...
+      - name: ⬆️ Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: output-${{ env.GitVersion_FullSemVer }}
+          path: output/*.*
+          compression-level: 0 # no compression
 
+      - id: set-version
+        name: Set version to output
+        run: echo "version=${{ env.GitVersion_FullSemVer }}" >> "$GITHUB_OUTPUT"
```
Comment (AlanRynne) [inline .github/workflows/main.yml:49]: Why the copy/paste? reusable workflows are ♥️
If we need some variation we can add inputs to it.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -9,10 +9,47 @@ on:
...
+        
   deploy-installers:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Comment (AlanRynne) [inline .github/workflows/main.yml:52]: This doesn't need to be windows at all 🙂

**Review by AlanRynne — APPROVED** [review]

**Comment by adamhathcock** [comment]
For https://spockle.atlassian.net/browse/DUI3-311

**Comment by adamhathcock** [comment]
DUI3-311

**Pull request merged**